RecoLocalCalo/HcalRecAlgos/interface/HcalDeterministicFit.h

```diff
@@ -62,4 +65,109 @@ class HcalDeterministicFit {
   
 };
 
+template<class Digi>
+void HcalDeterministicFit::apply(const CaloSamples & cs, const std::vector<int> & capidvec, const HcalCalibrations & calibs, const Digi & digi, std::vector<double> & HLTOutput) const {
+  std::vector<double> corrCharge;
+  std::vector<double> inputCharge;
+  std::vector<double> inputPedestal;
+  const unsigned int cssize = cs.size();
+  double gainCorr = 0;
+  for(unsigned int ip=0; ip<cssize; ++ip){
+    if( ip >= (unsigned) 10 ) continue; // Too many samples than what we wanna fit (10 is enough...) -> skip them
+    const int capid = capidvec[ip];
+    double charge = cs[ip];
+    double ped = calibs.pedestal(capid);
+    double gain = calibs.respcorrgain(capid);
+    gainCorr = gain;
+    inputCharge.push_back(charge);
+    inputPedestal.push_back(ped);
+  }
+
+  fPedestalSubFxn_.calculate(inputCharge, inputPedestal, corrCharge);
+
+  const HcalDetId& cell = digi.id();
+  double fpar0, fpar1;
+  if(TMath::Abs(cell.ieta())<16){
+    fpar0 = fparhb0;
+    fpar1 = fparhb1;
+  }else if(TMath::Abs(cell.ieta())==16||TMath::Abs(cell.ieta())==17){
```

Comment:
@xmniu thanks for the clarifications.
So, what about an array with the `ieta` of transition region?